### PR TITLE
DOC: add missing details to linalg.lstsq docstring

### PR DESCRIPTION
**Pull request by nilokr**
It turns out that `linalg.lstsq` also minimizes the 2-norm of `x` when `a` is rank-deficient. I found that by searching [the documentation for the LAPACK library](https://www.netlib.org/lapack/lug/node27.html), which is the current implementation of `lstsq`.

This PR rewrites the docstring for `linalg.lstsq` such that it contains details for all cases.

### CODE REVIEW

File: numpy/linalg/linalg.py [hunk 1]
```diff
@@ -2171,13 +2171,13 @@ def lstsq(a, b, rcond="warn"):
...
-    (i.e., the number of linearly independent rows of `a` can be less than,
-    equal to, or greater than its number of linearly independent columns).
-    If `a` is square and of full rank, then `x` (but for round-off error)
-    is the "exact" solution of the equation. Else, `x` minimizes the
-    Euclidean 2-norm :math:`|| b - a x ||`.
+    Computes the `n` element vector `x` that approximatively solves the equation
+    ``a @ x = b``, where `a` is an `m`-by-`n` matrix and `b` is a `m` element 
+    vector. When the system is overdetermined (``m >= n`` and ``rank(a) = n``), 
+    a solution `x` that minimizes the Euclidean 2-norm ``||b-ax||`` is found.
+    Else, when the system is underdetermined (``m < n`` and ``rank(A) = m``)
```
Comment (charris) [inline numpy/linalg/linalg.py:2178]: Maybe "If there are multiple minimizing solutions, the one with the smallest norm is returned". Note that the norm of the solution depends on column scaling (units) and choice of variables, so the smallest norm solution is seldom meaningful when the solution is not unique.

Comment (krnilo) [reply numpy/linalg/linalg.py]: Sounds better - I've updated my commit to reflect the suggestion, thanks!

The reason I stumbled on this smallest norm matter was that I was specifically looking for that solution in a rank-deficient problem. I was intrigued I got exactly what I wanted for free, so I did some digging and found why :)

File: numpy/linalg/linalg.py [hunk 2]
```diff
@@ -2171,13 +2171,14 @@ def lstsq(a, b, rcond="warn"):
...
 
-    Computes the vector x that approximatively solves the equation
+    Computes the vector `x` that approximatively solves the equation
     ``a @ x = b``. The equation may be under-, well-, or over-determined
     (i.e., the number of linearly independent rows of `a` can be less than,
     equal to, or greater than its number of linearly independent columns).
     If `a` is square and of full rank, then `x` (but for round-off error)
     is the "exact" solution of the equation. Else, `x` minimizes the
-    Euclidean 2-norm :math:`|| b - a x ||`.
+    Euclidean 2-norm :math:`||b-ax||`. If there are multiple minimizing
```
Comment (charris) [inline numpy/linalg/linalg.py:2180]: Keep spaces around `-` for readability.

Comment (charris) [reply numpy/linalg/linalg.py]: I fixed that.

**Comment by charris** [comment]
Thanks @krnilo .

**Comment by matthew-brett** [comment]
Worth pointing out that this is the behavior of the pseudoinverse?

**Pull request merged**